### DISCUSSION TITLE
Job posts new design.

### DISCUSSION THREAD

**Discussion by smoya** [bcneng/candebot#33] [Ideas]
I would like to see job posts to use Slack blocks layout. Also to include those fields the community consider are a must. 
Some people believe we need to add the option for specifying Bonuses or Stock options. 

I would start by suggesting the new template using blocks layout builder. See [here](https://app.slack.com/block-kit-builder/T2Y6L54R3#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Junior%20Backend%20Developer%20-%20*%3Cbcneng.org%7CBest%20Company%20ever%3E*%22%7D,%22accessory%22:%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Apply%22,%22emoji%22:true%7D,%22value%22:%22apply_button%22,%22url%22:%22https://bcneng.org%22,%22action_id%22:%22button-action%22%7D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22:label:%20*Type:*%20Employed%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:moneybag:%20*Salary:*%2055K%20-%2080K%20Eur%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:round_pushpin:%20*Location:*%20Remote%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:package:%20*Incentives:*%20Stock%20options%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:sports_medal:%20*Bonus:*%20Hire%20bonus%20-%208K%20Eur%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:raised_hands:%20*More%20info%20DM:*%20@candebot%22%7D%5D%7D%5D%7D)

An screenshot: 

![CleanShot 2021-06-10 at 18 06 59](https://user-images.githubusercontent.com/1083296/121559213-ae761e80-ca16-11eb-9540-4fe81952420b.png)

What do you all think? Please vote or add your suggestions in this discussion.
For voting, use the new button on the left of the title: 
![CleanShot 2021-06-10 at 18 36 18](https://user-images.githubusercontent.com/1083296/121563525-c780ce80-ca1a-11eb-95a8-b3fff8f1fd02.png)

**Comment by urielsalis** [comment]
Maybe crazy idea, but we could add a optional field with a link to the #corp- channel of the company if one exists? :D

**Reply by smoya** [reply]
We do not support #corp channels anymore, at least temporary. There is no official text saying that but that's the reality.

**Comment by urielsalis** [comment]
And kind of a nitpick, but I would order it a bit more

Start with type and location side by side
Follow Salary on its own if it can be centered
Then bonus and incentives side by side
and last more info DM + corp channel

**Reply by smoya** [reply]
Would you mind using this [generator](https://app.slack.com/block-kit-builder/T2Y6L54R3#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Junior%20Backend%20Developer%20-%20*%3Cbcneng.org%7CBest%20Company%20ever%3E*%22%7D,%22accessory%22:%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Apply%22,%22emoji%22:true%7D,%22value%22:%22apply_button%22,%22url%22:%22https://bcneng.org%22,%22action_id%22:%22button-action%22%7D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22:label:%20*Type:*%20Employed%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:moneybag:%20*Salary:*%2055K%20-%2080K%20Eur%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:round_pushpin:%20*Location:*%20Remote%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:package:%20*Incentives:*%20Stock%20options%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:sports_medal:%20*Bonus:*%20Hire%20bonus%20-%208K%20Eur%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22:raised_hands:%20*More%20info%20DM:*%20@candebot%22%7D%5D%7D%5D%7D) so you can represent what you mean? Thanks!

**Reply by urielsalis** [reply]
![image](https://user-images.githubusercontent.com/7349355/121658512-b038f400-caa1-11eb-8bdd-ee3e741fc8b0.png)
Just a reorder, so when you read it side to side related information is next to each other

**Reply by alorma** [reply]
Hi. I was going to say the same, thanks @smoya to point me to this discussion.

Current layout is hard to read on desktop (too large text) and mobile devices (too much text), what @urielsalis proposes (or any other ordered stuff) could be a great addition.

**Comment by masylum** [comment]
I like it in isolation, I'm not sure is going to get too crowded on the channel. I think you can get emoji-overloaded at some point.

**Reply by smoya** [reply]
Yeah, I'm concerned as well, but currently one has emojis. Not sure if it feels in that way right now. Votes wanted here!

**Reply by ignasi35** [reply]
Let's do it as suggested and iterate later...

**Reply by joaoqalves** [reply]
One thing that might help create a good trade-off is to add 

> Salary: 55-80K EUR + Incentives

and create a thread with the incentives explained.

**Comment by SantiMunoz** [comment]
Keeping oneliner offers helps the readability and simplicity, I would priorities that over the block layout

**Reply by smoya** [reply]
Just to be aligned, you mean you prefer the current design, right? 

Votes wanted here

**Reply by SantiMunoz** [reply]
Yes exactly, I would prefer the current design and potentially adding more fields if necessary

**Comment by jonatanfm** [comment]
Can we crate a temp channel #hiring-job-board-new with this new format so we can see both current and new version? also for future changes we can use it as testing channel for new job posts designs

**Reply by smoya** [reply]
That's technically doable 👍 . We could use it during a period of time and see if people prefers it over the current one.

**Comment by alexpdp7** [comment]
+1000 to "type", I think this is a very useful addition. I would keep bonus/incentives in a comment if the poster feels they are really remarkable. I +1 above the one who said preferred one-liner.

**Reply by alexpdp7** [reply]
Adding to my thread- I'd say there should be a free space with a limited amount of space to it. Put the most important things there, the rest can be seen by clicking on the link or reading the thread if you are interested...

**Comment by matiasburni** [comment]
we can add a field time-off (xxd|flexible|summer time|fridays out|etc)

**Reply by ignasi35** [reply]
That sounds like a better fit for a `perks` field with a bit of a lee-way. that is, instead of adding a new, specific field re `time-off` I think a new field `perks` where `time-off` and other relevant perks can be listed. 

Maybe the list should be checkboxes (or limited list of allowed words)  instead of a random text area.

**Comment by smoya** [comment]
@cristinaverdi Is adding a new field to the current (single line) format that include the nature of the post (Agency or final company) here https://github.com/bcneng/candebot/pull/36. That would be included in this final design.

**Comment by smoya** [comment]
I would like to also give special visibility to offers for Junior roles. 
Maybe enlightening somehow the text, adding a format, or whatever.
